fpga_interchange/README.md

```diff
@@ -0,0 +1,162 @@
+## FPGA interchange nextpnr architecture
+
+This nextpnr architecture is a meta architecture that in theory will implement
+any architecture that emits a complete FPGA interchange device database.
+
+### FPGA interchange
+
+The FPGA interchange is a set of file formats intended to describe any modern
+island based FPGA.  It consists of three primary file formats:
+
+ - Device database
+   - This is a description of a particular FPGA fabric.  This description
+     includes placement locations, placement constraints and a complete
+     description of the routing fabric.
+   - This file will also include timing information once added.
+
+ - Logical netlist
+   - This is the output of a synthesis tool.  This is equivalent to the
+     Yosys JSON format, EDIF, or eblif.
+   - As part of future nextpnr development, a frontend will be added that
+     takes this format as input.
+
+ - Physical netlist
+   - This is the output of a place and route tool. It can describe a clustered
+     design, a partially or fully placed design, and a partially or fully
+     routed design.
+
+### Current status
+
+This architecture implementation can be compiled in conjunction with a FPGA
+interchange device database, and the outputs from
+`fpga_interchange.nextpnr_emit`, which is part of the
+[python-fpga-interchange](https://github.com/SymbiFlow/python-fpga-interchange/)
+library.
+
+The current implementation is missing essential features for place and route.
+As these features are added, this implementation will become more useful.
+
+ - [ ] Placement constraints are unimplemented, meaning invalid or unroutable
+       designs can be generated from the placer.
+ - [ ] Logical netlist macro expansion is not implemented, meaning that any
+       macro primitives are unplacable.  Common macro primitives examples are
+       differential IO buffers (IBUFDS) and some LUT RAM (e.g. RAM64X1D).
+ - [ ] Cell -> BEL pin mapping is not in place, meaning any primitives that
+       have different BEL pins with respect to their cell pins will not be
+       routable.
+ - [ ] Nextpnr only allows for cell -> BEL pin maps that are 1 to 1.  The
+       FPGA interchange accomidates cell -> BEL pin maps that include 1 to
```

Comment:
nitpick - can you run a spelling check on this?

Reply:
Done!